tensorflow/c/kernels/ops/summary.cc

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/c/ops.h"
+#include "tensorflow/core/framework/selective_registration.h"
+#include "tensorflow/core/platform/logging.h"
+
+static void scalar_summary_shape_inference_fn(TF_ShapeInferenceContext* ctx,
+                                              TF_Status* status) {
+  TF_SetStatus(status, TF_OK, ""); 
+  TF_ShapeHandle* result = TF_NewShapeHandle();
+  // Make shape handle a scalar value (empty shape)
+  TF_ShapeInferenceContextSetOutput(ctx, 0, result, status);
```

Comment:
Just returning shape constructed with `TF_NewShapeHandle` seems different from current behavior.
`TF_NewShapeHandle` just [calls](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/c/ops.cc;l=104?q=tf_newshapehandle) empty `ShapeHandle` [constructor](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/framework/shape_inference.h;drc=a6207b8f2e5c62524392e1d2b22f87cfc3ec1f10;l=96). This means `ptr_` inside `ShapeHandle` is `nullptr` and `IsSet` would return false.

At the same time original registration creates a [scalar shape](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/framework/common_shape_fns.h;l=83;drc=a6207b8f2e5c62524392e1d2b22f87cfc3ec1f10). Clicking through, it does actually create a `Shape` object. So, it seems like `ptr_` would not be `nullptr` in `ShapeHandle`.

I am thinking we might have to add ` TF_ShapeInferenceContextScalar` similar to the vector shape function [here](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/c/ops.h;l=285). What do you think?